engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/dto/externaltask/FetchExternalTasksDto.java

```diff
@@ -318,21 +319,24 @@ protected SortMapper(List<SortingDto> sorting, FetchAndLockBuilder builder) {
      * Applies the sorting field mappings to the builder and returns it.
      */
     protected FetchAndLockBuilder getBuilderWithSortConfigs() {
-      sorting.forEach(dto -> {
-        fieldMappingKey(dto).ifPresent(key -> FIELD_MAPPINGS.get(key).accept(builder));
-        orderMappingKey(dto).ifPresent(key -> ORDER_MAPPINGS.get(key).accept(builder));
-      });
-
+      for (SortingDto dto : sorting) {
```

Comment:
🔧 IMHO, the logic "configure the builder based on this field or throw bad request exception" can be reused to simplify this function using Extract Method ([intellij](https://www.jetbrains.com/help/idea/extract-method.html), [eclipse](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2FgettingStarted%2Fqs-ExtractMethod.htm)).

Here is a git [patch](https://github.com/camunda/camunda-bpm-platform/files/14929232/chore.engine-rest.__Refactor_getBuilderWithSortConfigs__-_Extract_configureFieldOrBadReque.patch) with the above in case you'd like to use.

Reply:
Good idea, applied.